rules/TypeDeclaration/NodeAnalyzer/ClassMethodParamTypeCompleter.php

```diff
@@ -38,8 +38,13 @@ public function __construct(
     public function complete(ClassMethod $classMethod, array $classParameterTypes, int $maxUnionTypes): ?ClassMethod
     {
         $hasChanged = false;
+        $totalTypes = count($classParameterTypes);
 
         foreach ($classParameterTypes as $position => $argumentStaticType) {
+            if ($totalTypes > 1 && $argumentStaticType instanceof UnionType) {
+                return null;
+            }
```

Comment:
@internalsystemerror I just re-check that this verify is not valid, because multiple args no longer working now when one of them is UnionType, eg:

```php
final class NarrowUnion
{
    public function run(MethodCall $methodCall, StaticCall $staticCall, String_ $string)
    {
        $this->someExpr($methodCall, $staticCall);
        $this->someExpr($staticCall, $staticCall);
        $this->someExpr($string, $staticCall);
    }

    private function someExpr($expr, $staticCall)
    {
    }
}
```

above is now completely skipped, which actually should be:

```diff
-     private function someExpr($expr, $staticCall)
+    private function someExpr(\PhpParser\Node\Expr\MethodCall|\PhpParser\Node\Expr\StaticCall|\PhpParser\Node\Scalar\String_ $expr, \PhpParser\Node\Expr\StaticCall| $staticCall)
    {
    }
```

I will add separate PR for that :)